### PR TITLE
TST: XFAIL Travis read_html tests

### LINKED ISSUE #29622: Travis 3.6 Slow Build Test Failures - BS4 Issue
`test_thousands_macau_index_col` in `test_html` fails using beautifulsoup4 version 4.8.1 and passes using version 4.7.1.

Example failures:
https://travis-ci.org/pandas-dev/pandas/jobs/611742966

As pointed out by @jreback in this PR - https://github.com/pandas-dev/pandas/pull/29603

Example failing build pre my changes in #29513
https://travis-ci.org/pandas-dev/pandas/jobs/611347428

Attempting to debug issue and will report to bs4. Should we pin in the meantime to version <= 4.7.1?

### PR DESCRIPTION
**Pull request by alimcmaster1**
- [x] xref #29622

### CODE REVIEW

**Comment by jbrockmendel** [comment]
can't make it strict?

File: pandas/tests/io/test_html.py [hunk 1]
```diff
@@ -382,10 +382,12 @@ def test_thousands_macau_stats(self, datapath):
...
         assert not any(s.isna().any() for _, s in df.items())
 
-    # https://github.com/pandas-dev/pandas/issues/29622
-    @pytest.mark.xfail(reason='fails for bs4 version >= 4.8.0', strict=False)
     @pytest.mark.slow
     def test_thousands_macau_index_col(self, datapath):
+        # https://github.com/pandas-dev/pandas/issues/29622
+        if self.read_html.keywords.get("flavor") == "bs4":
+            pytest.xfail("fails for bs4 version >= 4.8.0")
+
```
Comment (jbrockmendel) [inline pandas/tests/io/test_html.py]: did you change this from the decorator in response to my comment?  if so, thats not what i meant.  the decorator is preferred, but without the `strict=False`

Comment (alimcmaster1) [reply pandas/tests/io/test_html.py]: Yeah just want to xfail one particular parameterization - `bs4` and the parameterization is at class scope - agree decorator is usually preferred.

Updated now - think its easiest to do this in the function itself for this case - unless you know a more elegant way?

Comment (alimcmaster1) [reply pandas/tests/io/test_html.py]: Just seen I can potentially go for the same approach as you did here https://github.com/pandas-dev/pandas/pull/30521/files?

Comment (jbrockmendel) [reply pandas/tests/io/test_html.py]: if the test doesnt use the class fixture(s), could it just be moved out of the class?

If not then the approach in 30521 would be my preference, yes.

Comment (alimcmaster1) [reply pandas/tests/io/test_html.py]: Some class fixtures are used so easier to keep where is. Updated - thanks!

**Comment by jbrockmendel** [comment]
looks like there was an xpass on the old_np build.  possibly related: i think bs4 had a release yesterday

**Comment by alimcmaster1** [comment]
> looks like there was an xpass on the old_np build. possibly related: i think bs4 had a release yesterday

Due to the fact we pin bs4 to 4.6.0 in the `py36_locale_slow_old_np` build. I've unpinned - let me know if this is not appropriate?

 Other option is to make this not strict?

**Comment by jbrockmendel** [comment]
> I've unpinned - let me know if this is not appropriate?

@datapythonista preference for this pin?

> Other option is to make this not strict?

If all else fails, yah.  I appreciate you putting in the effort to avoid this last resort.

**Comment by alimcmaster1** [comment]
Can confirm travis-36-slow build is now passing - https://travis-ci.org/pandas-dev/pandas/jobs/631033419?utm_medium=notification&utm_source=github_status

We can remove this from allowed failures in `.travis.yml` if desired

**Comment by datapythonista** [comment]
> @datapythonista preference for this pin?

We've got bs4 also pinned at 4.6.0 in the minimum versions build. I think ideally we should have a minimum versions build for the slow tests too. I guess if we had it, it'd fail like the old np build. So I guess unpinning will avoid the failure, but won't really fix the problem.

Do we want to increase the minimum required version instead?

**Comment by alimcmaster1** [comment]
> We've got bs4 also pinned at 4.6.0 in the minimum versions build. I think ideally we should have a > minimum versions build for the slow tests too. I guess if we had it, it'd fail like the old np build. So I > guess unpinning will avoid the failure, but won't really fix the problem.

Thanks for the info!

Should the minimum version build just also run the slow tests opposed to a sep build?

> Do we want to increase the minimum required version instead?

Okay shall we bump to min beautifulsoup4 == 4.8.0 ? (`azure-36-minimum_versions.yml` and `azure-36-locale_slow.yaml`) Therefore this xfail can be strict across all builds?

**Comment by datapythonista** [comment]
> Should the minimum version build just also run the slow tests opposed to a sep build?

My understanding is that a build with both the "normal" and the slow tests would be too slow. That's why we have the slow marker, so they are called in separate builds.



> Okay shall we bump to min beautifulsoup4 == 4.8.0 ?

This is from just few months ago. I think we're usually compatible with dependencies that are at least couple of years old.

File: ci/deps/azure-36-locale_slow.yaml [hunk 2]
```diff
@@ -13,7 +13,7 @@ dependencies:
   - pytest-azurepipelines
 
   # pandas dependencies
-  - beautifulsoup4==4.6.0
+  - beautifulsoup4=4.6.0
```
Comment (alimcmaster1) [inline ci/deps/azure-36-locale_slow.yaml]: This should be single =

**Comment by alimcmaster1** [comment]
> > Should the minimum version build just also run the slow tests opposed to a sep build?
> 
> My understanding is that a build with both the "normal" and the slow tests would be too slow. That's why we have the slow marker, so they are called in separate builds.
> 
> > Okay shall we bump to min beautifulsoup4 == 4.8.0 ?
> 
> This is from just few months ago. I think we're usually compatible with dependencies that are at least couple of years old.

Thanks @datapythonista for the info!

Okay so i've adjust the xfail to take into account the version of bs4 installed + removed the travis 36 slow build from the allowed failures.

Happy to create a min version build for the slow tests too as you suggested (sep PR). If that's a useful for us?

**Comment by datapythonista** [comment]
> Happy to create a min version build for the slow tests too as you suggested (sep PR). If that's a useful for us?

I think it makes more sense to discuss and add this as part of #29685. I don't think we really understand well what's being tested. I think creating builds from zero in an organized way is better than add this individually. I'm on it.

**Comment by alimcmaster1** [comment]
All green - @jbrockmendel / @datapythonista mind reviewing?

**Review by datapythonista — APPROVED** [review]
Didn't check the details, but if we can't fix the error properly, this looks good.

**Comment by jbrockmendel** [comment]
Thanks @alimcmaster1

**Pull request merged**